src/coreclr/src/jit/gentree.h

```diff
@@ -6687,6 +6688,25 @@ inline bool GenTree::IsIntegralConstVector(ssize_t constVal)
     return false;
 }
 
+//-------------------------------------------------------------------
+// IsSIMDZero: returns true if this this is a SIMD vector
+// with all its elements equal to zero.
+//
+// Returns:
+//     True if this represents an integral const SIMD vector.
+//
+inline bool GenTree::IsSIMDZero()
+{
+#ifdef FEATURE_SIMD
+    if ((gtOper == GT_SIMD) && (AsSIMD()->gtSIMDIntrinsicID == SIMDIntrinsicInit))
```

Comment:
Is there any need/desire to recognize `SIMDIntrinsicGetZero` or the equivalent `HWIntrinsic` kinds?

Reply:
`SIMDIntrinsicGetZero` becomes `SIMDIntrinsicInit` so that's covered, but we may want to consider the `HWIntrinsic` kinds if/when we unify their handling in codegen.